bam_to_mate_hist.py

```diff
@@ -254,38 +332,58 @@ def extract_stats(stat_list, bamfile, outfile_name, count_diff_refname_stub=Fals
     Returns:
         stat_dict ({str:float/str}: mappings of the stats and data suitable to be consumed by report generator
     '''
-    stat_dict = {}
+    # some path handling, doesn't need to happen in here really.
     stat_dict["BAM_FILE_PATH"] = os.path.split(bamfile)[-1]
-    num_pairs = stat_list[7]
-    stat_dict["NUM_PAIRS"] = num_pairs
-
-    print "Histograms written to:", os.path.abspath(outfile_name + "_long.png"), os.path.abspath(outfile_name + "_short.png")
     stat_dict["PATH_TO_LONG_HIST"] = os.path.abspath(outfile_name + "_long.png")
     stat_dict["PATH_TO_SHORT_HIST"] = os.path.abspath(outfile_name + "_short.png")
+    print "Histograms written to:", os.path.abspath(outfile_name + "_long.png"), os.path.abspath(outfile_name + "_short.png")
+
+    # only some things in dict get pretty floatified
+    to_props = ["ZERO_DIST_PAIRS",
+                "NUM_10KB_PAIRS",
+                "NUM_DIFF_CONTIG_PAIRS",
+                "NUM_SPLIT_READS",
+                "NUM_DUPE_READS",
+                "MAPQ0_READS"]
+
+    num_pairs = stat_dict["NUM_PAIRS"]
+
+    out_dict = est_proportions_pretty(stat_dict=stat_dict, stats=to_props)
+    # these two are calculated by read rather than by pair, correct
```

Comment:
You added a third property but left this saying 'these two'. Probably can just say "the following properties are calculated by read rather than read pair" so it is just a generic comment

Reply:
good point.